jdisc_http_service/src/main/java/com/yahoo/jdisc/http/filter/JdiscFilterRequest.java

```diff
@@ -113,6 +117,14 @@ public void setUserPrincipal(Principal principal) {
         this.parent.setUserPrincipal(principal);
     }
 
+    @Override
+    public List<X509Certificate> getClientCertificateChain() {
+        return Optional.ofNullable(parent.context().get(ServletRequest.JDISC_REQUEST_X509CERT))
+                .map(X509Certificate[].class::cast)
```

Comment:
Missing instanceof check, but I guess that can't happen.

Reply:
It want it to fail hard if it's not `X509Certificate[]`.